code/android-signal-library/src/main/java/com/adobe/marketing/mobile/signal/SignalHitProcessor.kt

```diff
@@ -0,0 +1,117 @@
+/*
+  Copyright 2022 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+package com.adobe.marketing.mobile.signal
+
+import androidx.annotation.VisibleForTesting
+import com.adobe.marketing.mobile.services.*
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+internal class SignalHitProcessor : HitProcessing {
+    private val networkService: Networking
+
+    companion object {
+        private const val LOG_TAG = "SignalHitProcessor"
+        private const val HIT_QUEUE_RETRY_TIME_SECONDS = 30
+    }
+
+    constructor() {
+        this.networkService = ServiceProvider.getInstance().networkService
+    }
+
+    @VisibleForTesting
+    internal constructor(networkService: Networking) {
```

Comment:
Maybe we can set it directly in ServiceProvider instead of having a constructor for it?

Reply:
We can't do that as `ServiceProvider` is in Core, another package.